RELEASE/scripts/autoscend/auto_util.ash

```diff
@@ -4326,6 +4326,31 @@ boolean haveAny(boolean[item] array)
 	return false;
 }
 
+boolean acquireOrPull(item it)
+{
+	//this function is for when you want to make sure you have 1 of an item
+	//if you have one it returns true. if you don't it will craft one. if it can't it will pull it.
+	
+	if(possessEquipment(it)) return true;
+	if(item_amount(it) > 0)  return true;
+	if(retrieve_item(1, it)) return true;
+	if(canPull(it))
+	{
+		if(pullXWhenHaveY(it, 1, 0)) return true;
+	}
+	
+	//special handling via pulling 1 ingredient to craft the item desired
+	if(it == $item[meteorb] && canPull($item[metal meteoroid]))
```

Comment:
I would expand the first part of the condition to include all the items the metal meteoroid can be made into.
e.g `$items[asteroid belt, meteorb, ...] contains it`
See https://kol.coldfront.net/thekolwiki/index.php/Metal_meteoroid

Reply:
Could you generalize this with `get_ingredients`?

```
> ash get_ingredients($item[asteroid belt])

Returned: aggregate int [item]
metal meteoroid => 1
```

Reply:
@Malibu-Stacey will do
@jeparo probably, but i am certain on how to implement it. As it's documentation states it  returns a map of the direct ingredients needed to craft something (so no multi step support). And it does not account for what you already have.

maybe a seperate function of "getMissingIngredients which takes `get_ingredients`, then adjusts it to remove ingredients you already have. then returns that. Then we count the total number of ingredients in map. And if it is just the 1 pull it?

That all being said. It feels like going beyond the scope of the current PR

Reply:
I agree this is out-of-scope.

Reply:
@malibu-stacey change made

@jeparo I just noticed I accidentally said `I am certain on how to implement it` was supposed to have a not in there. I am not certain.